src/torrent.cpp

```diff
@@ -6769,7 +6778,7 @@ namespace libtorrent
 			// TODO: 2 if peer is a really good peer, maybe we shouldn't disconnect it
 			// perhaps this logic should be disabled if we have too many idle peers
 			// (with some definition of idle)
-			if (peer && peer->peer_rank() < p->peer_rank())
+			if (peer != nullptr && peer->peer_rank() < p->peer_rank())
```

Comment:
no idea, didn't inspect that logic, just changed it to `!= nullptr` because it's a raw pointer